planner/scan.go

```diff
@@ -103,23 +115,32 @@ func (n *scanNode) Next() (bool, error) {
 		return false, nil
 	}
 
-	// keep scanning until we find a doc that passes the filter
-	for {
-		var err error
-		n.docKey, n.currentValue, err = n.fetcher.FetchNextDoc(n.p.ctx, n.documentMapping)
-		if err != nil {
-			return false, err
-		}
-		n.execInfo.docFetches++
+	var err error
+	n.docKey, n.currentValue, err = n.fetcher.FetchNextDoc(n.p.ctx, n.documentMapping)
+	if err != nil {
+		return false, err
+	}
+	// @question: Should this be after the length check?
+	n.execInfo.docFetches++
```

Comment:
answer: Represents the attempts at fetching docs. Since the Fetch call is before and we do this after ensuring it was not a failure, it's fine where it is at the moment.

Reply:
Commenting here for both this answer and your other comment.

Technically it means that `docFetches` and `filterMatches` will basically be the same value (or differ by at most 1).

Reply:
> Technically it means that `docFetches` and `filterMatches` will basically be the same value (or differ by at most 1).

I am going to talk from the perspective of what used to happen... The majority of the time, `docFetches` and `iterations` would be similar (because for every iteration we would attempt to fetch. But it's **not strictly** true. Also `docFetches` and `filterMatches` will mostly **NOT be same**, their difference is **not strictly** most by 1. It all depended on the number of docs and how many match the query or not. Consider these examples from our current execution explain tests.

## Consider scan node execute results...

### `TestExecuteExplainRequestWithOrderFieldOnParent`
```
"iterations":    uint64(3),
"docFetches":    uint64(3),
"filterMatches": uint64(2),
```

### `TestExecuteExplainMutationRequestWithCreate`
```
"iterations":    uint64(1),
"docFetches":    uint64(1),
"filterMatches": uint64(1),
```

### `TestExecuteExplainMutationRequestWithDeleteUsingFilter`
```
"iterations":    uint64(2),
"docFetches":    uint64(3),
"filterMatches": uint64(1),
```

### `TestExecuteExplainMutationRequestWithUpdateUsingFilter`
```
"iterations":    uint64(4),
"docFetches":    uint64(6),
"filterMatches": uint64(2),
```

### `TestExecuteExplainRequestWithDocumentsButNoMatches`
```
"iterations":    uint64(1),
"docFetches":    uint64(3),
"filterMatches": uint64(0),
```

For example in the last example it had two documents, it fetched 2 times and none matched the filter. The third time it fetched there was no more docs so total attempts are 3, with 0 matches, and we entered scan node once.

## Fetcher update in this PR
I have to read the implementation in more detail to see if it is still possible to differentiate between the `attempts fetches took place till match found` vs `total maches found`. Might even just need to return a variable with result from the fetcher call if we want to keep track similar to before of all the fetch attempts till match.

Reply:
My whole point is that because of the changes in this PR, the behavior is now different. Previously, the scanner "fetched" a document from the fetcher, then the scanner was responsible for filtering. 

One of the core changes in this PR is that we are moving the filtering *from* the scanner *to* the fetcher.

So keeping track of both `filterFetches` and `filterMatches` in the scanner is meaningless now.

Reply:
Yup makes sense, I did review the implementation in more depth and seems like in order to get that info now we would have to have the fetcher logic store that metric and then communicate back to the scanner. IMO let's keep the metrics we collect scoped to planner nodes only so the ones we need to bend over backward for in a more complicated manner "out-side" the plan nodes, I would skip those metrics for now.
tldr: can just remove/skip the metric that isn't as relevant now